iothub/service/src/Query/QueryClient.cs

```diff
@@ -0,0 +1,126 @@
+
+        private string _hostName;
+        private IotHubConnectionProperties _credentialProvider;
+        private HttpClient _httpClient;
+        private HttpRequestMessageFactory _httpRequestMessageFactory;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected QueryClient()
+        {
+        }
+
+        internal QueryClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _credentialProvider = credentialProvider;
+            _hostName = hostName;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Retrieves a handle through which a result for a given query can be fetched.
+        /// </summary>
+        /// <param name="sqlQueryString">The SQL query.</param>
+        /// <param name="pageSize">The maximum number of items per page.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A handle used to fetch results for a SQL query.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the provided SQL query string is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if the provided SQL query string is empty or whitespace.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-query-language"/>
+        public virtual IQuery CreateAsync(string sqlQueryString, int? pageSize = null, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Creating query", nameof(CreateAsync));
+            try
+            {
+                return new Query(async (token) => await ExecuteAsync(
+                    sqlQueryString,
+                    pageSize,
+                    token,
+                    cancellationToken));
+            }
+            catch (Exception ex)
+            {
+                Logging.Error(this, $"{nameof(CreateAsync)} threw an exception: {ex}", nameof(CreateAsync));
```

Comment:
Still elsewhere too.

```suggestion
                if (Logging.IsEnabled)
                    Logging.Error(this, $"{nameof(CreateAsync)} threw an exception: {ex}", nameof(CreateAsync));
```